cli/command/stack/list.go

```diff
@@ -18,18 +32,168 @@ func newListCommand(dockerCli command.Cli) *cobra.Command {
 		Short:   "List stacks",
 		Args:    cli.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if dockerCli.ClientInfo().HasKubernetes() {
-				kli, err := kubernetes.WrapCli(dockerCli, kubernetes.NewOptions(cmd.Flags()))
-				if err != nil {
-					return err
-				}
-				return kubernetes.RunList(kli, opts)
-			}
-			return swarm.RunList(dockerCli, opts)
+			return runList(cmd, dockerCli, opts)
 		},
 	}
 
 	flags := cmd.Flags()
 	flags.StringVar(&opts.Format, "format", "", "Pretty-print stacks using a Go template")
+	flags.StringSliceVar(&opts.Namespaces, "namespace", []string{"default"}, "Kubernetes namespaces to use")
+	flags.SetAnnotation("namespace", "kubernetes", nil)
+	flags.SetAnnotation("namespace", "experimentalCLI", nil)
```

Comment:
Can't this use `InstallFlags()` (which I would suggest should be renamed to `AddNamespaceFlag()`.

Reply:
It can't because here it's
```
flags.StringSliceVar(&opts.Namespaces, "namespace", []string{"default"}, "Kubernetes namespaces to use")
```
whereas in InstallFlags soon to be AddNamespaceFlag it's
```
flags.String("namespace", "default", "Kubernetes namespace to use")
```
e.g. most ```docker stack``` sub-commands accept only one ```--namespace``` but ```docker stack ls``` can handle a list of them.